x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -167,6 +168,105 @@ export function hasMoreAncestors(state: DataState): boolean {
   return tree ? resolverTreeModel.hasMoreAncestors(tree) : false;
 }
 
+/**
+ * A map of `entity_id`s to functions that provide information about
+ * related events by ECS `.category` Primarily to avoid having business logic
+ * in UI components.
+ */
+export const relatedEventInfoByEntityId = createSelector(
+  relatedEventsByEntityId,
+  relatedEventsStats,
+  function selectLineageLimitInfo(
+    /* eslint-disable no-shadow */
+    relatedEventsByEntityId,
+    relatedEventsStats
+    /* eslint-enable no-shadow */
+  ) {
+    const relatedEventInfoEntries = [...(relatedEventsStats || new Map()).entries()].map(
```

Comment:
how about:
```
// if there are no related event stats, there are no related event info objects either
if (!relatedEventStats) {
  return new Map()
}
const relatedEventInfoEntries = [relatedEventsStats.entries()].map(
  // ...  
```

This makes the guard logic immediately clear

Reply:
https://github.com/elastic/kibana/pull/70097/commits/f2f607b248e0eaa25cbdbdd8716983c86a55ca5c